doma-processor/src/test/resources/org/seasar/doma/internal/apt/processor/metamodel/ScopeTest_ItemEntity.txt

```diff
@@ -0,0 +1,64 @@
+
+    private final java.util.List<org.seasar.doma.jdbc.criteria.metamodel.PropertyMetamodel<?>> __allPropertyMetamodels;
+
+    public final org.seasar.doma.jdbc.criteria.metamodel.PropertyMetamodel<java.lang.Long> id = new org.seasar.doma.jdbc.criteria.metamodel.DefaultPropertyMetamodel<java.lang.Long>(java.lang.Long.class, __entityType, "id");
+
+    public final org.seasar.doma.jdbc.criteria.metamodel.PropertyMetamodel<java.lang.String> name = new org.seasar.doma.jdbc.criteria.metamodel.DefaultPropertyMetamodel<java.lang.String>(java.lang.String.class, __entityType, "name");
+
+    public final org.seasar.doma.jdbc.criteria.metamodel.PropertyMetamodel<java.time.LocalDateTime> publishedAt = new org.seasar.doma.jdbc.criteria.metamodel.DefaultPropertyMetamodel<java.time.LocalDateTime>(java.time.LocalDateTime.class, __entityType, "publishedAt");
+
+    private final org.seasar.doma.internal.apt.processor.entity.ItemScope __scope__org_seasar_doma_internal_apt_processor_entity_ItemScope = new org.seasar.doma.internal.apt.processor.entity.ItemScope();
+
+    public ItemEntity_() {
+        this("");
+    }
+
+    public ItemEntity_(String qualifiedTableName) {
+        this.__qualifiedTableName = java.util.Objects.requireNonNull(qualifiedTableName);
+        java.util.ArrayList<org.seasar.doma.jdbc.criteria.metamodel.PropertyMetamodel<?>> __list = new java.util.ArrayList<>(3);
+        __list.add(id);
+        __list.add(name);
+        __list.add(publishedAt);
+        __allPropertyMetamodels = java.util.Collections.unmodifiableList(__list);
+    }
+
+    @Override
+    public org.seasar.doma.jdbc.entity.EntityType<org.seasar.doma.internal.apt.processor.entity.ItemEntity> asType() {
+        return __qualifiedTableName.isEmpty() ? __entityType : new org.seasar.doma.jdbc.criteria.metamodel.EntityTypeProxy<>(__entityType, __qualifiedTableName);
+    }
+
+    @Override
+    public java.util.List<org.seasar.doma.jdbc.criteria.metamodel.PropertyMetamodel<?>> allPropertyMetamodels() {
+        return __allPropertyMetamodels;
+    }
+
+    public java.util.function.Consumer<org.seasar.doma.jdbc.criteria.declaration.WhereDeclaration> published() {
+        return __scope__org_seasar_doma_internal_apt_processor_entity_ItemScope.published(this);
+    }
+
+    public java.util.function.Consumer<org.seasar.doma.jdbc.criteria.declaration.OrderByNameDeclaration> newest() {
+        return __scope__org_seasar_doma_internal_apt_processor_entity_ItemScope.newest(this);
+    }
+
+    public java.util.function.Consumer<org.seasar.doma.jdbc.criteria.declaration.WhereDeclaration> idEq(java.lang.Long id) {
+        return __scope__org_seasar_doma_internal_apt_processor_entity_ItemScope.idEq(this, id);
+    }
+
+    public java.util.function.Consumer<org.seasar.doma.jdbc.criteria.declaration.WhereDeclaration> idEq(java.util.List<java.lang.Long> id) {
+        return __scope__org_seasar_doma_internal_apt_processor_entity_ItemScope.idEq(this, id);
+    }
```

Comment:
I'm in trouble because I can't think of an implementation that generates this method.
Can you help me?

If difficult, generics will not be supported.
The strategy to be implemented in the future may be good.

Reply:
It seems that we have to replace formal type parameters with actual type parameters.
The following code may be helpful.

https://github.com/domaframework/doma/blob/33a7a5131557d48301929bd5c8a8e021b06a2ee8/doma-processor/src/main/java/org/seasar/doma/internal/apt/decl/Declarations.java#L130

Reply:
Thank you! I was able to pass the test. 🎉